### DISCUSSION TITLE
Why you adding DB::connection()->getQueryGrammar() in getValue() funciton as parameter?

### DISCUSSION THREAD

**Discussion by zanysoft** [laravel/framework#48061] [Q&A]
### Laravel Version

10.18.0

### PHP Version

8.1.0

### Database Driver & Version

MariaDB 10.6.5

### Description
Whit reference: https://laravel.com/docs/10.x/upgrade#database-expressions

Why you adding `DB::connection()->getQueryGrammar()` in `getValue()` funciton as parameter? 

```php
use Illuminate\Support\Facades\DB;
 
$expression = DB::raw('select 1');
 
$string = $expression->getValue(DB::connection()->getQueryGrammar());

```

In your class [`Illuminate\Database\Query\Expression`](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Query/Expression.php) there is extra and uncalled/unused injection of `Illuminate\Database\Grammar`.

```php
namespace Illuminate\Database\Query;

use Illuminate\Contracts\Database\Query\Expression as ExpressionContract;
use Illuminate\Database\Grammar;

class Expression implements ExpressionContract
{
    /**
     * The value of the expression.
     *
     * @var string|int|float
     */
    protected $value;

    /**
     * Create a new raw query expression.
     *
     * @param  string|int|float  $value
     * @return void
     */
    public function __construct($value)
    {
        $this->value = $value;
    }

    /**
     * Get the value of the expression.
     *
     * @param  \Illuminate\Database\Grammar  $grammar
     * @return string|int|float
     */
    public function getValue(Grammar $grammar)
    {
        return $this->value;
    }
}
```

### Steps To Reproduce

https://laravel.com/docs/10.x/upgrade#database-expressions

**Comment by zanysoft** [comment]
I think laravel hired unexperienced developers.

**Reply by henzeb** [reply]
Some choices wouldn't be mine either. But I don't think they hired 'unexperienced' developers...
There must be a good reason for this. If anyone would elaborate on this, that 'd be great. @driesvints @nunomaduro @taylorotwell

**Comment by crynobone** [comment]
You can refer to https://github.com/laravel/framework/pull/47210 for some benefits
